### PR TITLE
Adds CatwalkStation

### PR DESCRIPTION
**Pull request by EuSouAFazer**
## About The Pull Request

Adds CatwalkStation into the game.
Also adds its associated arrival shuttle, cargo shuttle and escape shuttle.
Also adds a few new floor variants: coldroom openspace and breathable air shuttle hulls.

Full Catwalk FAQ, Feedback Central and Why It's Good for the Game: https://hackmd.io/@kNRZ9uxhShS3s2N9scOXxQ/BJgfF0JhR

## Changelog
:cl:
add: Adds CatwalkStation.
/:cl:
image

image
image

image
image
image
image

### CODE REVIEW

**Comment by EuSouAFazer** [comment]
Once this passes a basic review process I'll ask for a testmerge somewhere

File: code/datums/shuttles/emergency.dm [hunk 1]
```diff
@@ -265,6 +265,14 @@
 	description = "We pulled this one out of Mothball just for you!"
 	occupancy_limit = "40"
 
+
+/datum/map_template/shuttle/emergency/birdshot
```
Comment (Pepsilawn) [inline code/datums/shuttles/emergency.dm:269]: I love this new station called birdshot

Comment (EuSouAFazer) [reply code/datums/shuttles/emergency.dm]: Fixed

**Comment by BanementI** [comment]
image
poster museum on second z-level (feature)

**Comment by BanementI** [comment]
i know this should be in the feedback thread but i dont ever use TG forums.. large parts of maintenance don't have an area set, which results in these fullbright, unpowered sections
<details>
<summary>images</summary>

image

image
</details>

**Comment by EuSouAFazer** [comment]
FUCK

**Comment by MMMiracles** [comment]
<details>
  <summary>Noted Issues</summary>

**General Issues:**
You should probably do a pass on every room on wall equipment again. Noticed things like rooms with firelocks missing fire alarms or small things like light switches or intercoms. Some rooms (i.e bartender backroom) have a fire alarm but no firelock.

Please make a variant of the current catwalks with a delayed deconstruction step or something similar, as of right now I can only imagine the hell a single assistant with a wirecutter could cause in areas like Engineering. Antagonists/griefers are going to turn these secure areas into a spleef arena and I assume that isn't an intended side effect.

I'm 50/50 on the station's current aesthetic. On one hand there are areas I quite like (i.e the bridge area), but then you have other areas (i.e the main halls) that feel like I'm playing 2013 Boxstation again with how underdeveloped they feel. Little things like floor detailing go a long way and it seems like you already have a decent idea on how to use some of these more creatively (the simplified station map below the bridge as an example). 

**Specific Issues:**
image
Random stamp module by the courtroom


image
Overly dark corner in courtroom


image
Piping revealed on the glass in front of the brig, just wrap it around the glass


image
Misaligned railing above brig


image
Detective windoor doesn't allow anyone access (captain can't even open it)


image
Brig lockdown buttons have no associated access


image
image
image
image
All heads excluding the captain are missing key auths.
  

image
The cold in the kitchen freezer is setting off the fire alarm and also you rickrolled me fuck you


image
Anything that drops some sort of connection, whether it be sink or APC is gonna look ugly on these kinds of floors. Might want to use regular tiling here.


image
Kitchen in general feels a bit understocked compared to other maps. No enzyme, knife, less beakers, no dinnerware vendor, ect.


image
Why does the bartender get a centcom hat what


image
These mines can't be moved, they're structures.  Not gonna work like you think they will.


image
Missing waste pipe below Hydroponics


image
These endings on the catwalks look ugly with pipes IMO, might want to put some sort of cap-off at the ends to help it better blend together


image
This entire area is space.


image
Same with this area.


image
HOP area feels too dark.


image
Main bathroom area and upper area has zero lighting.


image
Used the wrong access on these doors, they're effectively all-access.


image
Maintenance area lacks an APC.


image
Vacant commissary has no way to 'secure' it with bolted doors like other maps, intentional?


image
image
This staircase leads into a wall.

image
Xenobiology has zero central lighting.


image
Cryo lacks a cooler.


image
Pharmacy/chemical storage uses an unnamed atmos airlock and has a locker blocking it by default.


image
Upper section between morgue and CMO's office is dark as hell.
</details>

This is about an hour just walking around and testing basic things like access and visibility. I would recommend you do something similar after getting through this list before looking into a testmerge. Larger projects like this can cause some serious tunnel vision and you'll end up overlooking things players might find obvious because they haven't been staring at the same map for the past few months.

**Comment by EuSouAFazer** [comment]
@MMMiracles Fixed the issues you posted except for the wires' end after catwalk one; that one is for coders that know way more about this stuff than me, sorry'. I don't know how to fix that.

Regarding harder to deconstruct catwalks: I'd like to see some playtesting before making calls like that. A big part of Catwalk is to introduce a different meta to the game, one where catwalks, jumping over gaps and multi-Z are big considerations for security and antagonists. Players carrying metal rods or wirecutters for bonus mobility in the catwalks is interesting emergent gameplay and intended - stuff like long fall boots is present on the security armory for this reason and I can see the dolphin gloves being very popular in this map. 

I'll see what I'll be able to do regarding the hallways and de-plaining a few rooms.

**Comment by Admiral-Gigaton** [comment]
Wow this is actually real? Just been seeing it appear in the coding threads of discord, didnt know it was actually worked on still

**Comment by EuSouAFazer** [comment]
Did a number of fixes + new screenshots on the opening post!

**Pull request closed without merging**